client/settings/fraud-protection/advanced-settings/utils.ts

```diff
@@ -0,0 +1,353 @@
+					ruleConfiguration?.min_amount?.toString() ?? ''
+				)
+					? {
+							key: Checks.CHECK_ORDER_TOTAL,
+							operator: CheckOperators.OPERATOR_LT,
+							value: buildFormattedRulePrice(
+								ruleConfiguration?.min_amount?.toString() ?? ''
+							),
+					  }
+					: {
+							key: Checks.CHECK_ORDER_TOTAL,
+							operator: CheckOperators.OPERATOR_GT,
+							value: buildFormattedRulePrice(
+								ruleConfiguration?.max_amount?.toString() ?? ''
+							),
+					  };
+			}
+			break;
+	}
+
+	return ruleBase;
+};
+
+const findCheck = (
+	current: FraudProtectionSettingsCheck,
+	checkKey: string,
+	operator: string
+): FraudProtectionSettingsCheck | boolean => {
+	if (
+		isFraudProtectionSettingsSingleCheck( current ) &&
+		checkKey === current.key &&
+		operator === current.operator
+	) {
+		return current;
+	}
+
+	if (
+		! isFraudProtectionSettingsSingleCheck( current ) &&
+		current?.checks
+	) {
+		for ( const i in current.checks ) {
+			const check = current.checks[ i ];
+			const result = findCheck( check, checkKey, operator );
+
+			if ( false !== result ) {
+				return result;
+			}
+		}
+	}
+
+	return false;
+};
+
+export const writeRuleset = (
+	config: ProtectionSettingsUI
+): AdvancedFraudProtectionSettings[] => {
+	const rulesetConfig = [];
+	for ( const key in config ) {
+		if ( config[ key ].enabled ) {
+			rulesetConfig.push(
+				buildRuleset( key, config[ key ].block, config[ key ] )
+			);
+		}
+	}
+
+	return rulesetConfig.filter( ( rule ) => rule );
+};
+
+export const readRuleset = (
+	rulesetConfig: AdvancedFraudProtectionSettings[] | string
+): ProtectionSettingsUI => {
+	const defaultUIConfig = {
+		[ Rules.RULE_ADDRESS_MISMATCH ]: { enabled: false, block: false },
+		[ Rules.RULE_INTERNATIONAL_IP_ADDRESS ]: {
+			enabled: false,
+			block: false,
+		},
+		[ Rules.RULE_IP_ADDRESS_MISMATCH ]: {
+			enabled: false,
+			block: false,
+		},
+		[ Rules.RULE_ORDER_ITEMS_THRESHOLD ]: {
+			enabled: false,
+			block: false,
+			min_items: null,
+			max_items: null,
+		},
+		[ Rules.RULE_PURCHASE_PRICE_THRESHOLD ]: {
+			enabled: false,
+			block: false,
+			min_amount: null,
+			max_amount: null,
+		},
+	};
+	const parsedUIConfig = {} as ProtectionSettingsUI;
+
+	if ( 'string' !== typeof rulesetConfig ) {
+		for ( const id in rulesetConfig ) {
+			const rule = rulesetConfig[ id ];
+
+			switch ( rule.key ) {
+				case Rules.RULE_ADDRESS_MISMATCH:
+					parsedUIConfig[ rule.key ] = {
+						enabled: true,
+						block: rule.outcome === Outcomes.BLOCK,
+					};
+					break;
+				case Rules.RULE_INTERNATIONAL_IP_ADDRESS:
+					parsedUIConfig[ rule.key ] = {
+						enabled: true,
+						block: rule.outcome === Outcomes.BLOCK,
+					};
+					break;
+				case Rules.RULE_IP_ADDRESS_MISMATCH:
+					parsedUIConfig[ rule.key ] = {
+						enabled: true,
+						block: rule.outcome === Outcomes.BLOCK,
+					};
+					break;
+				case Rules.RULE_ORDER_ITEMS_THRESHOLD:
+					const minItems = findCheck(
```

Comment:
Maybe we should also rename these variables with a `Check` suffix, like `minItemsCheck`, just to increase clarity.

Reply:
That's a good idea! I've made that change in [484a4ec](https://github.com/Automattic/woocommerce-payments/pull/6032/commits/484a4ec68b48aeb62e34fa4806572dd0983c5cb8).